torch/distributed/_composable/fully_shard.py

```diff
@@ -96,7 +99,7 @@ def fully_shard(
     _register_root_pre_forward_hook(state, module)  # prepend last
     for submodule in module.modules():
         if (
-            submodule not in state._ignored_modules
```

Comment:
Users may pass 'ignored_parameters' instead of 'ignored_modules', in this case, it is not proper to use 'state._ignored_modules' here.

Reply:
If we ignore _some_ parameters from a module, then what should the semantics be? Should we still annotate it with `fully_shard`? Should we only annotate it with `fully_shard` if we ignored _all_ of the module's parameters (`recurse=False`)?

Reply:
good catch! I think we should insert the state for the sharded module even it has partial managed parameter and partial ignored parameters. 

As commented below, probably we will just use '_fully_sharded_module_to_handles' to decide whether we should insert states for sharded modules or not